### PR TITLE
Implemented endpoint to get torrents in a specific discovered channel

### PR DESCRIPTION
**Pull request by devos50**

### CODE REVIEW

**Comment by devos50** [comment]
```
test_channel_search (Tribler.Test.test_remote_search.TestRemoteChannelSearch) ... Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 798, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/endpoint.py", line 250, in dispersythread_data_came_in
    u"standalone_ep")
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/dispersy.py", line 1464, in on_incoming_packets
    (str(packet[0]), repr(packet[1])) for packet in packets]
exceptions.AssertionError: [('<Candidate 118.45.11.204:29104>', "'A\\x00B\\xa1\\x02r\\xf3\\xe0\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x8ac\\x00\\x00'")]
```

I think it's bit overkill to assert when a malformed (or corrupt?) package comes in. Maybe we should just ignore it.

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
```
Tribler.Test.Core.Modules.RestApi.test_channels_endpoints.TestChannelTorrentsEndpoint.test_get_torrents_in_channel (from nosetests)
timeout expired before end of test (5.000000 s.)
```

This is so weird, I specify a timeout 10 seconds...

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@lfdversluis ready for your review (this is the last one for today I promise :D)

File: doc/Tribler REST API.md [hunk 1]
```diff
@@ -79,6 +80,26 @@ Returns all discovered channels in Tribler.
...
+
+### Example response
+
+```json
+{
+    "torrents": [{
+        "id": 4,
+        "infohash": "97d2d8f5d37e56cfaeaae151d55f05b077074779",
+        "name": "Ubuntu-16.04-desktop-amd64",
+        "length": 8592385,
```
Comment (lfdversluis) [inline doc/Tribler REST API.md]: in bytes? May be unclear.

File: doc/Tribler REST API.md [hunk 2]
```diff
@@ -79,6 +80,26 @@ Returns all discovered channels in Tribler.
...
+        "category": "other",
+        "num_seeders": 42,
+        "num_leechers": 184
```
Comment (lfdversluis) [inline doc/Tribler REST API.md]: Isn't there time since last check item? These numbers change often!

File: doc/Tribler REST API.md (hunk 1 above)
Comment (devos50) [inline doc/Tribler REST API.md]: True, I've added a description about the unit of the torrent size. I also renamed the `length` key to `size` which I think is more clear.

File: doc/Tribler REST API.md (hunk 2 above)
Comment (devos50) [inline doc/Tribler REST API.md]: It turns out we have this information available in the database so I've added it to the response.

**Comment by devos50** [comment]
@lfdversluis I've processed your feedback, @whirm ready for your review.

File: Tribler/Core/Modules/restapi/channels_endpoint.py [hunk 3]
```diff
@@ -153,25 +146,83 @@ def render_DELETE(self, request):
...
+            "num_seeders": 42,
+            "num_leechers": 184,
+            "last_tracker_check": 1463176959
         }, ...]
     }
     """
 
+    def __init__(self, session, cid):
+        BaseChannelsEndpoint.__init__(self, session)
+        self.session = session
```
Comment (whirm) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: the superclass is already setting this attr

**Comment by whirm** [comment]
Other than that minor remark it looks fine. Squash!

**Comment by devos50** [comment]
@whirm ready for merge

**Pull request merged**